code/modules/paperwork/photocopier.dm

```diff
@@ -239,6 +243,71 @@
 	return p
 
 
+
+
+//////Premium(upgraded) photocopier/////
+
+
+
+
+/obj/structure/machinery/photocopier/wyphotocopier
+	name = "photocopier"
+	icon = 'icons/obj/structures/machinery/library.dmi'
+	icon_state = "bigscannerpro"
+	desc = "A photocopier used for copying...you know, photos! Also useful for copying documents in paper. This specific model have been manufactured by Weyland-Yutani in a more modern and robust frame than the average photocopiers you see being sold by smaller companies, while also carrying some of the most advanced technologies on the area of paper-printing, such as bigger toner economy and much higher printing capabilities. All that makes it the favorite among consumers that need to print high amounts of paperwork on the performance of their daily duties."
+	anchored = TRUE
+	density = TRUE
+	use_power = USE_POWER_IDLE
+	idle_power_usage = 50
+	active_power_usage = 300
+	power_channel = POWER_CHANNEL_EQUIP
+	copies = 1 //how many copies to print!
+	toner = 180 //how much toner is left! woooooo~
+	maxcopies = 30 //how many copies can be copied at once- idea shamelessly stolen from bs12's copier!
+
+
+/obj/structure/machinery/photocopier/wyphotocopier/attackby(obj/item/O as obj, mob/user as mob)
+	if(istype(O, /obj/item/paper))
+		if(!copy && !photocopy && !bundle)
+			if(user.drop_inv_item_to_loc(O, src))
+				copy = O
+				to_chat(user, SPAN_NOTICE("You insert the paper into \the [src]."))
+				flick("bigscannerpro1", src)
+				updateUsrDialog()
+		else
+			to_chat(user, SPAN_NOTICE("There is already something in \the [src]."))
+	else if(istype(O, /obj/item/photo))
+		if(!copy && !photocopy && !bundle)
+			if(user.drop_inv_item_to_loc(O, src))
+				photocopy = O
+				to_chat(user, SPAN_NOTICE("You insert the photo into \the [src]."))
+				flick("bigscannerpro1", src)
+				updateUsrDialog()
+		else
+			to_chat(user, SPAN_NOTICE("There is already something in \the [src]."))
+	else if(istype(O, /obj/item/paper_bundle))
+		if(!copy && !photocopy && !bundle)
+			if(user.drop_inv_item_to_loc(O, src))
+				bundle = O
+				to_chat(user, SPAN_NOTICE("You insert the bundle into \the [src]."))
+				flick("bigscannerpro1", src)
+				updateUsrDialog()
+	else if(istype(O, /obj/item/device/toner))
+		if(toner == 0)
+			if(user.temp_drop_inv_item(O))
+				qdel(O)
+				toner = 180
+				to_chat(user, SPAN_NOTICE("You insert the toner cartridge into \the [src]."))
+				updateUsrDialog()
+		else
+			to_chat(user, SPAN_NOTICE("This cartridge is not yet ready for replacement! Use up the rest of the toner."))
+	else if(HAS_TRAIT(O, TRAIT_TOOL_WRENCH))
+		playsound(loc, 'sound/items/Ratchet.ogg', 25, 1)
+		anchored = !anchored
+		to_chat(user, SPAN_NOTICE("You [anchored ? "wrench" : "unwrench"] \the [src]."))
+	return
```

Comment:
Is there any actual difference here other than the toner amount? If not, this can be done in a less wordy way

Reply:
The sprite flick have been changed too, from "bigscanner1" to the special "bigscannerpro1"

Reply:
If the only difference is the flick toner and flick, you can just add say `var/animation_state = "bigscanner1"` on `/obj/structure/machinery/photocopier` have all the flicks on the base type use `animation_state` as the argument instead, and then have `/obj/structure/machinery/photocopier/wyphotocopier` override it to be `animation_state = "bigscannerpro1"`